velox/exec/MergeJoin.cpp

```diff
@@ -1060,6 +1060,29 @@ RowVectorPtr MergeJoin::doGetOutput() {
     return nullptr;
   }
 
+  auto firstNonNullIndex = firstNonNull(rightInput_, rightKeyChannels_);
```

Comment:
Shall we just revert the relevant change in https://github.com/facebookincubator/velox/pull/13039

Reply:
@xiaoxmeng No, The issue in this PR can be resolved by reverting #13039. However, a bug still persists in PR #13039. We need to verify whether the `prepareOutput ` function returns true. If it does, we should return the output. Otherwise, we must set the rows to null for the range `[0, firstNotNullIndex)`.